Ibistic.Public.NSwag.Extensions/Ibistic.Public.NSwag.Extensions/ApiVersionQueryStringProcessor.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Namotion.Reflection;
+using NSwag.Generation.Processors.Contexts;
+using System.Collections;
+using System.Linq;
+using System.Reflection;
+using NSwagProcessors = NSwag.Generation.Processors;
+
+namespace Ibistic.Public.NSwag.Extensions
+{
+    public class ApiVersionQueryStringProcessor : NSwagProcessors.IOperationProcessor
+    {
+        /// <summary>Processes an action controller trying to add its api version to the action query string path.</summary>
+        /// <param name="context">The processor context.</param>
+        /// <returns>true if the action should be added to the OpenApi specification.</returns>
+        public bool Process(OperationProcessorContext context)
+        {
+            var nswagApiVersionProcessor = context.Settings.OperationProcessors.TryGet<NSwagProcessors.ApiVersionProcessor>();
+            string[] filteredVersions = nswagApiVersionProcessor.IncludedVersions;
+
+            string[] actionApiVersions = GetActionApiVersions(context);
+
+            // If the action/controller has no the ApiVersion attribute
+            if (actionApiVersions.Length == 0)
+            {
+                // Include action in the OpenApi specification if the document is not being created with filtered versions.
+                return filteredVersions == null || filteredVersions.Length == 0;
+            }
+
+            // If no filter by versions, get the first (higher) one
+            if (filteredVersions == null || filteredVersions.Length == 0)
+            {
+                IncludeApiVersionInQueryString(context, actionApiVersions[0]);
```

Comment:
As far as I can see, actionApiVersions is not sorted. How do you know that [0] is the highest one?

Reply:
The GetActionApiVersions method gets firstly the attributes from the action and then for the controller using the Concat function so the first one is the one to be applied.

Reply:
Yes, but what if you have versions on an action like this:
[ApiVersion("1.1")]
[ApiVersion("2.1")]
[ApiVersion("1.0")]

Should you not sort them in some way?

Reply:
Modified code to get highest api version of an action and modified Readme file saying that ApiVersion attributes must be ordered from higher to lower.